docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,107 @@
+
+Scale the _weight_ of the uptime check based on time elapsed since last contact.
+
+The previous uptime reputation system worked much like the current audit reputation. See [Node Selection](node-selection.md) for more information.
+> α(n) = λ·α(n-1) + _w_(1+_v_)/2
+>
+> β(n) = λ·β(n-1) + _w_(1-_v_)/2
+>
+> R(n) = α(n) / (α(n) + β(n))
+
+In the case of a successful uptime check, _v_ is set to 1. Otherwise, it is set to -1. Previously, _w_, or _weight_, was configured at runtime and thereafter remained static. Thus, all uptime checks carried the same weight. However, should this really be the case?
+For example, a node fails two uptime checks within a timespan of 1 minute. The second failure does not give us very much additional information does it? We already knew the node was offline less than one minute ago. If the difference were one hour that information might be more useful in determining uptime.
+
+We might be able to account for this by adjusting _w_ based on how much time has elapsed since last contact.
+To do this we'll need a new configurable `Uptime Scale`: a duration against which to scale the weight of an individual uptime check. To demonstrate, let's take an Uptime Scale of one hour. If a node is found to be offline when audited and its last contact is greater than or equal to one hour ago, the weight is scaled to 100%. This check will result in the full reputation punishment. A second failed check occurs 30 minutes later. This is one half the scale: the weight is reduced by 50%. Another failed check occurs one minute later, and the weight is scaled back to 1/60 the punishment.
```

Comment:
> If a node is found to be offline when audited and its last contact is greater than or equal to one hour ago, the weight is scaled to 100%.

That is a wrong assumtion. If the last contact is more than one hour ago that doesn't mean the node was offline for one hour. It only means the node is offline now. We have to assume it was going offline a few seconds ago. Only if we keep pinging the node in an acceptable short interval we can count the difference between the first failed ping and the last failed ping as offline. Because storage nodes should checkin on startup we can add that as offline time as well.

and to be clear I mean pings not audits. Sending 2 audits per day and failing both could again mean the node was just  offline for a few seconds in both cases. We need to proof that it was offline the hole time. We can't just assume that.

Reply:
I didn't intend to imply that this idea is attempting to measure downtime. It's not. True, this alternate proposal and the previous system can only show that a node was offline during a single point in time. 
This is just an attempt to mitigate the previous system's capability of damaging a node's uptime reputation in a short period.

However, I think your point is relevant to the document's main design. The currently implemented downtime tracking does include a certain amount of assumption. It pings nodes whose last contact was a failure, and if it fails the ping, the time between the two failed contacts is assumed to be downtime and written into the downtime table.

The degree of confidence we can have in this assumption depends on the length of time between the two failed pings.

Reply:
Heh, I see that the title of this section actually does imply that. Your comment below has convinced me that the intent of downtime DQ really is about whether we believe this node will continue to cost us for repair. For that purpose, length of time offline is the best indicator.

Also, I'm now aware that, because a node will contact the satellite on start up, actually we can be certain that if a node fails two pings it was down the entire time.